### ISSUE TITLE
Imputation Problem Workflow

### ISSUE THREAD

**Issue by wasay464** [thuml/Time-Series-Library#694]
Hi. Thanks a lot for contributing this library to open source. Kudos to open source. I have a question regarding imputation task. I have a dataset with 0.1% of missing blocks in the data but they are at random timestamps. My questions are:

1) What is the workflow for performing imputation task in this situation? What should be my mask rate? Is it zero?
2) My data is sampled on a half-hourly interval. What changes should I need for imputation task?
3) What would be my test data and train data as I only have to perform imputation and there are no values to compare. So what does loss represent in this case.
4) A basic question. Suppose my data has random 800 missing values. How can i map the missing values back to the timeseries since my sequence length is 96. Can you please explain it. I am sorry for asking this basic question.

Again thanks a lot for your wonderful work. Keep it up! Best regards

**Comment by wuhaixu2016** [comment]
Hi, thanks for providing such detailed questions.
1. The current experiment pipeline will only calculate the loss on the masked data points (https://github.com/thuml/Time-Series-Library/blob/main/exp/exp_imputation.py#L124), thus I think you can try the mask ratio as 0.1% at the beginning. However, I think you might have some concerns about the exact training samples. You can also try the mask ratio as 10% during training but test it on the 0.1% missing situation.

2. You can try the embedding type as `m` (https://github.com/thuml/Time-Series-Library/blob/main/run.py#L37)

3. Generally speaking, you cannot obtain the model performance on the inference stage. Thus, I would like to suggest you try to split train and validation sets, where you can evaluate your model on the valid set.

4. just split the data into 8 segments where each segment has 100 time points.

Hope these answers are helpful to you.
